### ISSUE TITLE
Implement Batchnorm on OpenCL

### ISSUE THREAD

**Issue by leonardoaraujosantos** [leonardoaraujosantos/DLMatFramework#32]
## Introduction
Implement batchnorm (incuding spatial version) on OpenCL

## References
* [OpenCL version from caffe](https://github.com/BVLC/caffe/blob/opencl/src/caffe/greentea/cl_kernels/batch_norm.cl)
